drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md

```diff
@@ -0,0 +1,132 @@
+
+Fortunately Spring Boot embraces the Twelve Factor guidelines. Spring Boot supports configuration via environment variables out of the box and it's trivial to implement your own.
+
+Looking at the [example app](https://github.com/kabisa/dokku-boot-demo) I prepared there are two important things to notice:
+
+&nbsp;1. I've created two property aliases to allow configuration via `$PORT` and `$DATABASE_URL`. This is not strictly necessary but I prefer the more generic environment variables over the Spring specific ones.
+
+```java
+// application.properties
+spring.datasource.url=${database.url:jdbc:postgresql://localhost:5432/dokku-demo}
+server.port=${port:8080}
+```
+
+&nbsp;2. I've added a Procfile to instruct Dokku how to start our app. Dokku will read this file during deployment and use the `web` process to boot our app. You can read more about Procfiles [here](https://devcenter.heroku.com/articles/procfile).
+
+```sh
+# Procfile
+web: env DATABASE_URL=$JDBC_DATABASE_URL target/dokku-boot-demo-0.0.1-SNAPSHOT.jar
+```
+
+During deployment Dokku invokes `mvn install` on our app so it will build a jar. Here we instruct Dokku that to launch our app it needs to execute the jar. We set the `$DATABASE_URL` to `$JDBC_DATABASE_URL` which is an environment variable provided by the Java buildpack that formats the database connection string for JDBC compatibility.
+
+## 4. Deploy your app
+
+Alright so now our app is Dokku compatible. Before we can push our app to Dokku we need to configure our application in Dokku. For simplicity enter the following `dokku` commands in an SSH session on your server. Later you can setup the [Dokku CLI client](http://dokku.viewdocs.io/dokku/community/clients/) to run these commands directly from your local machine.
+
+First we'll create our app in Dokku:
+
+```sh
+$ dokku apps:create my-app
+```
+
+Next we'll install the Dokku Postgres plugin:
+
+```sh
+$ dokku plugin:install https://github.com/dokku/dokku-postgres.git postgres
+```
+
+This will pull in a Postgres Docker image so give it some time to complete.
+
+Next create a Postgres instance for our app:
+
+```sh
+$ dokku postgres:create my-app-postgres
+```
+
+Link it to our app:
+
+```sh
+$ dokku postgres:link my-app-postgres my-app
+```
+
+That's all the configuration we need, let's actually deploy our app now!
+
+First clone the example app to your local machine:
+
+```sh
+$ git clone https://github.com/kabisa/dokku-boot-demo.git
+$ cd dokku-boot-demo
+```
+
+Dokku uses a Heroku-style `git push to deploy` model, so we'll need to add a Git remote to our repo to point to Dokku's Git repo. Replace `dokku.example.org` with your actual Dokku hostname.
+
+```sh
+$ git remote add dokku dokku@dokku.example.org:my-app
+```
+
+And finally push our app to Dokku:
+
+```sh
+$ git push dokku
```

Comment:
Ah yes good point. `git push dokku` might not working depending on your `push.default` git config. Will change it to `git push dokku master` that'll always work.